File: 162-find-peak-element/find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
    //    int n = nums.size();
    //     for(int i=0; i<n-1; i++){
    //         if(nums[i] > nums[i+1]){
    //             return i;
    //         }
    //     }
    //     return n-1;
    // return max_element(nums.begin(),nums.end())-nums.begin();
    int start=0;
    int n=nums.size();
    int end=n-1;
    while(start<end){
        int mid=start+(end-start)/2;
        if(nums[mid]<nums[mid+1]){
            start=mid+1;
        }
        else{
            end=mid;
        }
    }
        return start;
    
    }
};

File: 162-find-peak-element/find-peak-element.java
class Solution {
    public int findPeakElement(int[] arr) {
    //       int n=arr.length;
    //       int start=1;
    //       int end=n-2;
    //       while(start<=end){
    //         int mid=start+(end-start)/2;
    //         if(arr[mid]>arr[mid-1]&&arr[mid]>arr[mid+1]){
    //             return mid;
    //         }
    //         else if(arr[mid-1]<arr[mid]){
    //     start=mid+1;
    //    }
    //         else{
    //             end=mid-1;
    //         }
    //       }
    //       return -1;
        

        
    int start=0;
    int end=arr.length-1;
    while(start<end){
        int mid=start+(end-start)/2;
        if(arr[mid]>arr[mid+1]){
            end=mid;
        }
        else{
            start=mid+1;
        }
    }
    return start;

    }
}